services/121-service/src/registration/registrations.controller.ts

```diff
@@ -376,32 +377,42 @@ export class RegistrationsController {
         programId,
         PermissionEnum.RegistrationAttributeFinancialUPDATE,
       );
+    const hasUpdateFspPermission =
+      await this.registrationsPaginateService.userHasPermissionForProgram(
+        userId,
+        programId,
+        PermissionEnum.RegistrationFspUPDATE,
+      );
 
-    if (!hasRegistrationUpdatePermission && !hasUpdateFinancialPermission) {
+    if (
+      !hasRegistrationUpdatePermission &&
+      !hasUpdateFinancialPermission &&
+      !hasUpdateFspPermission
+    ) {
       const errors = `User does not have permission to update attributes`;
       throw new HttpException({ errors }, HttpStatus.FORBIDDEN);
     }
 
     const partialRegistration = updateRegistrationDto.data;
 
-    if (!hasUpdateFinancialPermission && hasRegistrationUpdatePermission) {
-      for (const attributeKey of Object.keys(partialRegistration)) {
-        if (
-          FinancialAttributes.includes(attributeKey as keyof RegistrationEntity)
-        ) {
+    for (const attributeKey of Object.keys(partialRegistration)) {
+      if (
+        FinancialAttributes.includes(attributeKey as keyof RegistrationEntity)
+      ) {
+        if (!hasUpdateFinancialPermission) {
           const errors = `User does not have permission to update financial attributes`;
           throw new HttpException({ errors }, HttpStatus.FORBIDDEN);
         }
-      }
-    }
-
-    if (hasUpdateFinancialPermission && !hasRegistrationUpdatePermission) {
-      for (const attributeKey of Object.keys(partialRegistration)) {
-        if (
-          !FinancialAttributes.includes(
-            attributeKey as keyof RegistrationEntity,
-          )
-        ) {
+      } else if (
+        attributeKey ===
+        GenericRegistrationAttributes.programFinancialServiceProviderConfigurationName
+      ) {
+        if (!hasUpdateFspPermission) {
```

Comment:
- Should we also have a test that covers this?

Reply:
added